Sources/SecureXPC/Server/XPCServer.swift

```diff
@@ -350,11 +350,52 @@ public class XPCServer {
         }
     }
     
-    private func handleError(_ error: Error,
-                             connection: xpc_connection_t,
-                             reply: inout xpc_object_t?) {
+    // MARK: Error handling
+    
+    /// Wrapper around an error handling closure to ensure there's only ever one error handler regardless of whether it's synchronous or asynchronous.
+    fileprivate enum ErrorHandler {
+        case none
+        case sync((XPCError) -> Void)
+        case async((XPCError) async -> Void)
+        
+        func handle(_ error: XPCError) {
+            switch self {
+                case .none:
+                    ()
+                case .sync(let handler):
+                    handler(error)
+                case .async(let handler):
+                    if #available(macOS 10.15, *) {
+                        Task {
```

Comment:
Is spawning a new task the correct semantics here?

Reply:
I believe so. My understanding is if the current context is async then it'll attach to the TaskGroup under execution. Otherwise it'll cause a new one to be created. That behavior seems desirable.